crates/sui-indexer-alt/migrations/2024-11-02-160143_kv_epochs/up.sql

```diff
@@ -0,0 +1,56 @@
+-- Information related to an epoch that is available when it starts
+CREATE TABLE IF NOT EXISTS kv_epoch_starts
+(
+    epoch                       BIGINT        PRIMARY KEY,
+    protocol_version            BIGINT        NOT NULL,
+
+    -- Inclusive checkpoint lowerbound of the epoch.
+    cp_lo                       BIGINT        NOT NULL,
```

Comment:
there is no `tx_lo` here, is it to avoid DB read, any known tradeoff if so?

Reply:
You can always calculate `tx_lo` for an epoch as `tx_hi` of its predecessor. It's unlikely we would ever prune the epochs table, but if we did, we could make this strategy robust to that by keeping one extra row of `kv_epoch_ends`.